addons/microsoft_calendar/tests/test_update_events.py

```diff
@@ -1314,3 +1316,60 @@ def test_forbid_simple_event_become_recurrence_sync_on(self, mock_patch):
 
         # Assert that no patch call was made due to the recurrence update forbiddance.
         mock_patch.assert_not_called()
+
+    @patch.object(MicrosoftCalendarService, 'get_events')
+    @patch.object(MicrosoftCalendarService, 'delete')
+    @patch.object(MicrosoftCalendarService, 'insert')
+    def test_changing_event_organizer_to_another_user(self, mock_insert, mock_delete, mock_get_events):
+        """
+        Allow editing the event organizer to another user only if the proposed organizer have its Odoo Calendar synced.
+        The current event is deleted and then recreated with the new organizer.
+        An event with organizer as user A (self.organizer_user) will have its organizer changed to user B (self.attendee_user).
+        """
+        # Create event with organizer as user A and only the organizer as attendee.
+        self.assertTrue(self.env['calendar.event'].with_user(self.attendee_user)._check_microsoft_sync_status())
+        self.simple_event_values['user_id'] = self.organizer_user.id
+        self.simple_event_values['partner_ids'] = [Command.set([self.organizer_user.partner_id.id])]
+        event = self.env['calendar.event'].with_user(self.organizer_user).create(self.simple_event_values)
+
+        # Deactivate user B's calendar synchronization. Try changing the event organizer to user B.
+        # A ValidationError must be thrown because user B's calendar is not synced.
+        self.attendee_user.microsoft_synchronization_stopped = True
+        with self.assertRaises(ValidationError):
+            event.with_user(self.organizer_user).write({'user_id': self.attendee_user.id})
+
+        # Activate user B's calendar synchronization and try again without listing user B as an attendee.
+        # Another ValidationError must be thrown.
+        self.attendee_user.microsoft_synchronization_stopped = False
+        with self.assertRaises(ValidationError):
+            event.with_user(self.organizer_user).write({'user_id': self.attendee_user.id})
+
+        # Set mock return values for the event re-creation.
+        existing_events = self.env["calendar.event"].search([])
```

Comment:
Even though you're not expected to have many events in your DB, searching on everything might be a bit too much. You can either do a _search_read to get all the calendar_event ids (and get the last one) and then browse it, or you can do something like:

```suggestion
        new_event = self.env["calendar.event"].search([("id", ">", event.id)])
```